src/Ops.Service/EmediaService.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Ocuda.Ops.Service.Abstract;
+using Ocuda.Ops.Service.Filters;
+using Ocuda.Ops.Service.Interfaces.Promenade.Repositories;
+using Ocuda.Ops.Service.Interfaces.Promenade.Services;
+using Ocuda.Ops.Service.Models;
+using Ocuda.Promenade.Models.Entities;
+using Ocuda.Utility.Exceptions;
+
+namespace Ocuda.Ops.Service
+{
+    public class EmediaService : BaseService<EmediaService>, IEmediaService
+    {
+        private readonly IEmediaRepository _emediaRepository;
+        private readonly ICategoryRepository _categoryRepository;
+        private readonly IEmediaCategoryRepository _emediaCategoryRepository;
+
+        public EmediaService(ILogger<EmediaService> logger,
+            IHttpContextAccessor httpContextAccessor,
+            IEmediaRepository emediaRepository,
+            IEmediaCategoryRepository emediaCategoryRepository,
+            ICategoryRepository categoryRepository)
+            : base(logger, httpContextAccessor)
+        {
+            _emediaRepository = emediaRepository
+                ?? throw new ArgumentNullException(nameof(emediaRepository));
+            _categoryRepository = categoryRepository
+                ?? throw new ArgumentNullException(nameof(categoryRepository));
+            _emediaCategoryRepository = emediaCategoryRepository
+                ?? throw new ArgumentNullException(nameof(emediaCategoryRepository));
+        }
+
+        public async Task<ICollection<Emedia>> GetAllEmedia()
+        {
+            var emedias = (await _emediaRepository.GetAllAsync()).ToList();
+            var emediaCats = await _emediaCategoryRepository.GetAllAsync();
+            foreach (var emedia in emedias)
+            {
+                var emediaCategories = new List<Category>();
+                foreach (var emediacat in emediaCats.Where(_=>_.EmediaId == emedia.Id))
+                {
+                    emediaCategories.Add(await _categoryRepository.FindAsync(emediacat.CategoryId));
+                }
+                emedia.Categories = emediaCategories;
+            }
+            return emedias;
+        }
+
+        public async Task<Emedia> GetByStubAsync(string emediaStub)
+        {
+            var emedia = _emediaRepository.GetByClass(emediaStub);
```

Comment:
Are we looking up by stub or CSS class here? It doesn't seem like there would be a case why we would be looking it up by CSS class.

Reply:
I had forgotten to rename the function. Its searching by stub!